nifi-api/src/test/java/org/apache/nifi/registry/TestVariableRegistry.java

```diff
@@ -0,0 +1,126 @@
+
+        Map<String,String> variables = registry.getVariables();
+        assertTrue(variables.size() == 2);
+        assertTrue(variables.get("fake.property.1").equals("fake test value"));
+        assertTrue(registry.getVariableValue("fake.property.2").equals("fake test value"));
+    }
+
+    @Test
+    public void testReadProperties(){
+        Properties properties = new Properties();
+        properties.setProperty("fake.property.1","fake test value");
+        VariableRegistry registry = VariableRegistryFactory.getInstance(properties);
+        Map<String,String> variables = registry.getVariables();
+        assertTrue(variables.get("fake.property.1").equals("fake test value"));
+    }
+
+    @Test
+    public void testReadFiles(){
+        final Path fooPath = Paths.get("src/test/resources/TestVariableRegistry/foobar.properties");
+        final Path testPath = Paths.get("src/test/resources/TestVariableRegistry/test.properties");
+        VariableRegistry registry = VariableRegistryFactory.getInstance(fooPath.toFile(),testPath.toFile());
+        Map<String,String> variables = registry.getVariables();
+        assertTrue(variables.size() == 3);
+        assertTrue(variables.get("fake.property.1").equals("test me out 1"));
+        assertTrue(variables.get("fake.property.3").equals("test me out 3, test me out 4"));
+    }
+
+    @Test
+    public void testReadPaths(){
+        final Path fooPath = Paths.get("src/test/resources/TestVariableRegistry/foobar.properties");
+        final Path testPath = Paths.get("src/test/resources/TestVariableRegistry/test.properties");
+        VariableRegistry registry = VariableRegistryFactory.getInstance(fooPath,testPath);
+        Map<String,String> variables = registry.getVariables();
+        assertTrue(variables.size() == 3);
+        assertTrue(variables.get("fake.property.1").equals("test me out 1"));
+        assertTrue(variables.get("fake.property.3").equals("test me out 3, test me out 4"));
+    }
+
+    @Test
+    public void testAddRegistry(){
+
+        final Map<String,String> variables1 = new HashMap<>();
+        variables1.put("fake.property.1","fake test value");
+
+
+        final Path fooPath = Paths.get("src/test/resources/TestVariableRegistry/foobar.properties");
+        VariableRegistry pathRegistry = VariableRegistryFactory.getInstance(fooPath);
+
+        final Path testPath = Paths.get("src/test/resources/TestVariableRegistry/test.properties");
+        VariableRegistry fileRegistry = VariableRegistryFactory.getInstance(testPath.toFile());
+
+        Properties properties = new Properties();
+        properties.setProperty("fake.property.5","test me out 5");
+        VariableRegistry propRegistry = VariableRegistryFactory.getInstance(properties);
+
+        propRegistry.addRegistry(pathRegistry);
+        propRegistry.addRegistry(fileRegistry);
+
+        Map<String,String> variables = propRegistry.getVariables();
+        assertTrue(variables.size() == 4);
```

Comment:
Actually based on my updated comment on the map size (see above) yes this would fail if we based sizing on distinct keys.  I think originally size was based on actually entries (in Query.java implementation).